nestkernel/stimulating_backend.h

```diff
@@ -0,0 +1,179 @@
+  *
+  * @ingroup NESTio
+  */
+  virtual void enroll( StimulatingDevice&, const DictionaryDatum& ){};
+
+  /**
+   * Disenroll a `StimulatingDevice` from the `StimulatingBackend`.
+   *
+   * This function is considered to be the opposite of enroll() in the
+   * sense that it cancels the enrollment of a RecordingDevice from a
+   * RecordingBackend by deleting all device specific data. When
+   * setting a new recording backend for a recording device, this
+   * function is called for each backend the device is not enrolled
+   * with.
+   *
+   * @param device the RecordingDevice to be disenrolled
+   *
+   * @see enroll()
+   *
+   * @ingroup NESTio
+   */
+  virtual void disenroll( StimulatingDevice& ){};
+
+  /**
+   * Initialize global backend-specific data structures.
+   *
+   * This function is called on each backend right at the very beginning of
+   * `SimulationManager::run()`. It can be used for initializations which have
+   * to be repeated at the beginning of every single call to run in a
+   * prepare-run-run-...-run-run-cleanup sequence.
+   *
+   * @see post_run_hook()
+   *
+   * @ingroup NESTio
+   */
+  virtual void pre_run_hook() = 0;
+
+  /**
+   * Clean up the backend at the end of a Run.
+   *
+   * This is called right before `SimulationManager::run()` terminates. It
+   * allows the backend to flush open files, write remaining data to the
+   * screen, or perform similar operations that make sure that the user
+   * has access to all data from the previous simulation run.
+   *
+   * @see pre_run_hook()
+   *
+   * @ingroup NESTio
+   */
+  virtual void post_run_hook() = 0;
+
+  /**
+   * Do work required at the end of each simulation step.
+   *
+   * This is called at the very end of each simulation step. It can for example
+   * be used to carry out writing to files in a synchronized way, all threads
+   * on all MPI processes performing it at the same time.
+   *
+   * @see pre_run_hook()
+   *
+   * @ingroup NESTio
+   */
+  virtual void post_step_hook() = 0;
+
+  virtual void initialize() = 0;
+  virtual void finalize() = 0;
+
+  /**
+   * Prepare the backend at the beginning of the NEST Simulate function.
+   *
+   * This function is called by `KernelManager::prepare()` and allows the
+   * backend to open files, establish network connections, etc.
+   *
+   * @see cleanup()
+   *
+   * @ingroup NESTio
+   */
+  virtual void prepare() = 0;
+
+  /**
+  * Clean up the backend at the end of a user level call to the NEST Simulate
+  * function.
+  *
+  * This function is called by `SimulationManager::cleanup()` and allows the
+  * backend to close open files, close network connections, etc.
+  *
+  * @see prepare()
+  *
+  * @ingroup NESTio
+  */
+  virtual void cleanup() = 0;
+
+  void clear( const StimulatingDevice& ){};
+
+  static const std::vector< Name > NO_DOUBLE_VALUE_NAMES;
+  static const std::vector< Name > NO_LONG_VALUE_NAMES;
+};
+
+} // namespace
+
+#endif // STIMULATING_BACKEND_H
```

Comment:
All Doxygen comments are _still_ (more or less) just copies of the ones from `recording_backend.h`. Even the term `recording` is not replaced everywhere, which makes me doubt that the documentation is correct or appropriate like it is.

As an example, the documentation still talks about the non-existent function `set_value_names()` and the two static members don't make any sense to me in this class. Please update the Doxygen comments to reflect reality, and describe the interface of this class instead of just replacing terms. Thanks!

Reply:
The documentation of the functions is updated for the stimulated backend.